### PR TITLE
Allow using ittnotify (Intel Inspector)

### PR DESCRIPTION
**Pull request by masterleinad**
This pull request allows finding `ittnotify` (Intel Inspector). Related to #2375.

### CODE REVIEW

File: cmake/KokkosCore_config.h.in [hunk 1]
```diff
@@ -11,7 +11,6 @@
 #cmakedefine KOKKOS_ENABLE_THREADS
 #cmakedefine KOKKOS_ENABLE_CUDA
 #cmakedefine KOKKOS_ENABLE_HPX
-#cmakedefine KOKKOS_ENABLE_HWLOC
```
Comment (masterleinad) [inline cmake/KokkosCore_config.h.in]: This was duplicated.

**Review by dhollman — APPROVED** [review]
I assume this also applies the proper link flags? If so, LGTM

**Comment by dalg24** [comment]
> I assume this also applies the proper link flags?

It will link against libittnotify, that's all.  Are you referring to other flags?

**Comment by masterleinad** [comment]
I checked that `ittnotify` is required when configuring with `-DKokkos_ENABLE_DEBUG=ON` and using an `Intel` compiler and that I can compile the library and run the test suite successfully.

**Pull request merged**